.github/workflows/my-pipeline.yml

```
@@ -0,0 +1,42 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+  pull_request:
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker build --target test --tag my-test-image .
+      - run: docker run --env-file .env.test my-test-image:latest
+  docker-push-and-deploy:
+    name: Push to Docker Hub and Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and tag images
+        run: |-
+          docker build --target production --tag confuzzled/my-todo-app:latest .
+          docker tag confuzzled/my-todo-app:latest confuzzled/my-todo-app:$GITHUB_SHA
```

Minor, but you can apply both tags at the same time during the build stage:
```suggestion
          docker build --target production --tag confuzzled/my-todo-app:latest --tag confuzzled/my-todo-app:$GITHUB_SHA .
```